### ISSUE TITLE
Android build example app occur bazel not found the package

### ISSUE THREAD

**Issue by ggjj321** [google/mediapipe#3481]
OS: Ubuntu 18.04
bazel version:
Bazelisk version: v1.12.0
Build label: 5.2.0
Android
sdk: 30.0.3
ndk: 21.1.7075529

I'm trying to build a handtrackinggp example app. But "error: package com.android.dx.command.dexer does not exist" occur.
Please help me solve the problem.

There is the whole error message:
```
ERROR: /home/nvidia/.cache/bazel/_bazel_nvidia/ff4425722229fc486cc849b5677abe3f/external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/BUILD:3:13: Building external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/libdexer.jar (15 source files) and running annotation processors (AutoValueProcessor) failed: (Exit 1): java failed: error executing command external/remotejdk11_linux/bin/java -XX:-CompactStrings '--add-exports=jdk.compiler/com.sun.tools.javac.api=ALL-UNNAMED' '--add-exports=jdk.compiler/com.sun.tools.javac.main=ALL-UNNAMED' ... (remaining 17 arguments skipped)
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/DexBuilder.java:20: error: package com.android.dx.command.dexer does not exist
import com.android.dx.command.dexer.DxContext;
                                   ^
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/Dexing.java:16: error: package com.android.dx.cf.direct does not exist
import com.android.dx.cf.direct.DirectClassFile;
                               ^
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/Dexing.java:17: error: package com.android.dx.cf.direct does not exist
import com.android.dx.cf.direct.StdAttributeFactory;
                               ^
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/Dexing.java:18: error: package com.android.dx.command.dexer does not exist
import com.android.dx.command.dexer.DxContext;
                                   ^
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/Dexing.java:19: error: package com.android.dx.dex does not exist
import com.android.dx.dex.DexOptions;
                         ^
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/Dexing.java:20: error: package com.android.dx.dex.cf does not exist
import com.android.dx.dex.cf.CfOptions;
                            ^
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/Dexing.java:21: error: package com.android.dx.dex.cf does not exist
import com.android.dx.dex.cf.CfTranslator;
                            ^
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/Dexing.java:22: error: package com.android.dx.dex.code does not exist
import com.android.dx.dex.code.PositionList;
                              ^
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/Dexing.java:23: error: package com.android.dx.dex.file does not exist
import com.android.dx.dex.file.ClassDefItem;
                              ^
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/Dexing.java:24: error: package com.android.dx.dex.file does not exist
import com.android.dx.dex.file.DexFile;
                              ^
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/Dexing.java:25: error: package com.android.dx.util does not exist
import com.android.dx.util.ByteArray;
                          ^
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/DexBuilder.java:209: error: cannot find symbol
      DxContext context,
      ^
  symbol:   class DxContext
  location: class DexBuilder
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/Dexing.java:187: error: cannot find symbol
  private final DxContext context;
                ^
  symbol:   class DxContext
  location: class Dexing
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/Dexing.java:188: error: cannot find symbol
  private final DexOptions dexOptions;
                ^
  symbol:   class DexOptions
  location: class Dexing
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/Dexing.java:189: error: cannot find symbol
  private final CfOptions cfOptions;
                ^
  symbol:   class CfOptions
  location: class Dexing
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/Dexing.java:195: error: cannot find symbol
  public Dexing(DxContext context, DexingOptions options) {
                ^
  symbol:   class DxContext
  location: class Dexing
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/Dexing.java:200: error: cannot find symbol
  Dexing(DxContext context, DexOptions dexOptions, CfOptions cfOptions) {
         ^
  symbol:   class DxContext
  location: class Dexing
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/Dexing.java:200: error: cannot find symbol
  Dexing(DxContext context, DexOptions dexOptions, CfOptions cfOptions) {
                            ^
  symbol:   class DexOptions
  location: class Dexing
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/Dexing.java:200: error: cannot find symbol
  Dexing(DxContext context, DexOptions dexOptions, CfOptions cfOptions) {
                                                   ^
  symbol:   class CfOptions
  location: class Dexing
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/Dexing.java:206: error: cannot find symbol
  public static DirectClassFile parseClassFile(byte[] classfile, String classfilePath) {
                ^
  symbol:   class DirectClassFile
  location: class Dexing
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/Dexing.java:214: error: cannot find symbol
  public DexFile newDexFile() {
         ^
  symbol:   class DexFile
  location: class Dexing
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/Dexing.java:218: error: cannot find symbol
  public ClassDefItem addToDexFile(DexFile dest, DirectClassFile classFile) {
                                   ^
  symbol:   class DexFile
  location: class Dexing
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/Dexing.java:218: error: cannot find symbol
  public ClassDefItem addToDexFile(DexFile dest, DirectClassFile classFile) {
                                                 ^
  symbol:   class DirectClassFile
  location: class Dexing
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/Dexing.java:218: error: cannot find symbol
  public ClassDefItem addToDexFile(DexFile dest, DirectClassFile classFile) {
         ^
  symbol:   class ClassDefItem
  location: class Dexing
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/Dexing.java:131: error: cannot find symbol
    public CfOptions toCfOptions(DxContext context) {
                                 ^
  symbol:   class DxContext
  location: class DexingOptions
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/Dexing.java:131: error: cannot find symbol
    public CfOptions toCfOptions(DxContext context) {
           ^
  symbol:   class CfOptions
  location: class DexingOptions
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/Dexing.java:145: error: cannot find symbol
    public DexOptions toDexOptions() {
           ^
  symbol:   class DexOptions
  location: class DexingOptions
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/DexConverter.java:16: error: package com.android.dx.dex.file does not exist
import com.android.dx.dex.file.DexFile;
                              ^
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/DexConverter.java:29: error: cannot find symbol
  public DexFile toDexFile(byte[] classfile, String classfilePath) {
         ^
  symbol:   class DexFile
  location: class DexConverter
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/DexFileAggregator.java:19: error: package com.android.dex does not exist
import com.android.dex.Dex;
                      ^
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/DexFileAggregator.java:20: error: package com.android.dx.command.dexer does not exist
import com.android.dx.command.dexer.DxContext;
                                   ^
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/DexFileAggregator.java:21: error: package com.android.dx.merge does not exist
import com.android.dx.merge.CollisionPolicy;
                           ^
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/DexFileAggregator.java:22: error: package com.android.dx.merge does not exist
import com.android.dx.merge.DexMerger;
                           ^
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/DexFileAggregator.java:47: error: cannot find symbol
  private final ArrayList<Dex> currentShard = new ArrayList<>();
                          ^
  symbol:   class Dex
  location: class DexFileAggregator
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/DexFileAggregator.java:51: error: cannot find symbol
  private final DxContext context;
                ^
  symbol:   class DxContext
  location: class DexFileAggregator
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/DexFileArchive.java:18: error: package com.android.dex does not exist
import com.android.dex.Dex;
                      ^
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/DexLimitTracker.java:16: error: package com.android.dex does not exist
import com.android.dex.Dex;
                      ^
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/DexLimitTracker.java:17: error: package com.android.dex does not exist
import com.android.dex.FieldId;
                      ^
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/DexLimitTracker.java:18: error: package com.android.dex does not exist
import com.android.dex.MethodId;
                      ^
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/DexLimitTracker.java:19: error: package com.android.dex does not exist
import com.android.dex.ProtoId;
                      ^
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/DexLimitTracker.java:20: error: package com.android.dex does not exist
import com.android.dex.TypeList;
                      ^
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/DexFileAggregator.java:61: error: cannot find symbol
      DxContext context,
      ^
  symbol:   class DxContext
  location: class DexFileAggregator
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/DexFileAggregator.java:79: error: cannot find symbol
  public DexFileAggregator add(Dex dexFile) {
                               ^
  symbol:   class Dex
  location: class DexFileAggregator
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/DexFileAggregator.java:131: error: cannot find symbol
  private void writeMergedFile(Dex... dexes) {
                               ^
  symbol:   class Dex
  location: class DexFileAggregator
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/DexFileAggregator.java:144: error: cannot find symbol
  private Dex merge(Dex... dexes) throws IOException {
                    ^
  symbol:   class Dex
  location: class DexFileAggregator
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/DexFileAggregator.java:144: error: cannot find symbol
  private Dex merge(Dex... dexes) throws IOException {
          ^
  symbol:   class Dex
  location: class DexFileAggregator
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/DexLimitTracker.java:47: error: cannot find symbol
  public boolean track(Dex dexFile) {
                       ^
  symbol:   class Dex
  location: class DexLimitTracker
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/DexLimitTracker.java:58: error: cannot find symbol
  private void trackFieldsAndMethods(Dex dexFile) {
                                     ^
  symbol:   class Dex
  location: class DexLimitTracker
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/DexLimitTracker.java:69: error: cannot find symbol
  private static String typeName(Dex dex, int typeIndex) {
                                 ^
  symbol:   class Dex
  location: class DexLimitTracker
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/DexLimitTracker.java:94: error: cannot find symbol
    static MethodDescriptor fromDex(Dex dex, int methodIndex) {
                                    ^
  symbol:   class Dex
  location: class MethodDescriptor
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/DexLimitTracker.java:75: error: cannot find symbol
    static FieldDescriptor fromDex(Dex dex, int fieldIndex) {
                                   ^
  symbol:   class Dex
  location: class FieldDescriptor
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/DexFileArchive.java:47: error: cannot find symbol
  public DexFileArchive addFile(ZipEntry entry, Dex dex) throws IOException {
                                                ^
  symbol:   class Dex
  location: class DexFileArchive
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/DexFileAggregator.java:188: error: cannot find symbol
  private class RunDexMerger implements Callable<Dex> {
                                                 ^
  symbol:   class Dex
  location: class DexFileAggregator
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/DexFileAggregator.java:190: error: cannot find symbol
    private final Dex[] dexes;
                  ^
  symbol:   class Dex
  location: class DexFileAggregator.RunDexMerger
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/DexFileAggregator.java:192: error: cannot find symbol
    public RunDexMerger(Dex... dexes) {
                        ^
  symbol:   class Dex
  location: class DexFileAggregator.RunDexMerger
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/DexFileAggregator.java:197: error: cannot find symbol
    public Dex call() throws IOException {
           ^
  symbol:   class Dex
  location: class DexFileAggregator.RunDexMerger
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/DexFileAggregator.java:212: error: cannot find symbol
    private final ListenableFuture<Dex> dex;
                                   ^
  symbol:   class Dex
  location: class WriteFile
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/DexFileAggregator.java:216: error: cannot find symbol
    public WriteFile(String filename, ListenableFuture<Dex> dex, DexFileArchive dest) {
                                                       ^
  symbol:   class Dex
  location: class WriteFile
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/DexFileMerger.java:21: error: package com.android.dex does not exist
import com.android.dex.Dex;
                      ^
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/DexFileMerger.java:22: error: package com.android.dex does not exist
import com.android.dex.DexFormat;
                      ^
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/DexFileMerger.java:23: error: package com.android.dx.command.dexer does not exist
import com.android.dx.command.dexer.DxContext;
                                   ^
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/DexFileSplitter.java:20: error: package com.android.dex does not exist
import com.android.dex.Dex;
                      ^
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/DexFileSplitter.java:21: error: package com.android.dex does not exist
import com.android.dex.DexFormat;
                      ^
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/DexFiles.java:16: error: package com.android.dex does not exist
import com.android.dex.Dex;
                      ^
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/DexFiles.java:17: error: package com.android.dx.dex.file does not exist
import com.android.dx.dex.file.DexFile;
                              ^
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/DexFiles.java:28: error: cannot find symbol
  public static Dex toDex(DexFile dex) throws IOException {
                          ^
  symbol:   class DexFile
  location: class DexFiles
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/DexFiles.java:28: error: cannot find symbol
  public static Dex toDex(DexFile dex) throws IOException {
                ^
  symbol:   class Dex
  location: class DexFiles
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/DexFiles.java:35: error: cannot find symbol
  static byte[] encode(DexFile dex) throws IOException {
                       ^
  symbol:   class DexFile
  location: class DexFiles
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/DexFileMerger.java:164: error: cannot find symbol
      defaultValue = "" + DexFormat.MAX_MEMBER_IDX,
                          ^
  symbol:   variable DexFormat
  location: class Options
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/DexFileSplitter.java:112: error: cannot find symbol
      defaultValue = "" + DexFormat.MAX_MEMBER_IDX,
                          ^
  symbol:   variable DexFormat
  location: class Options
Target //mediapipe/examples/android/src/java/com/google/mediapipe/apps/handtrackinggpu:handtrackinggpu failed to build
Use --verbose_failures to see the command lines of failed build steps.
```

**Comment by kuaashish** [comment]
Hi @ggjj321, Please refer this issue [#3365(comment)](https://github.com/google/mediapipe/issues/3365#issuecomment-1135883896) for your query. Thank you!

**Comment by ggjj321** [comment]
Thanks for your answer

**Comment by kuaashish** [comment]
Hi @ggjj321, Could you please confirm this is still an issue or resolved as per the above [comment.](https://github.com/google/mediapipe/issues/3481#issuecomment-1173366927) Thank you!

**Comment by tianhonglilei** [comment]
@kuaashish Hello, Sorry to bother you, I added the code, but it doesn't work. I tried several times.

**Comment by kuaashish** [comment]
Hi @tianhonglilei, Could you please share the complete error log and raise a new separate issue with complete steps to reproduce. Thank you!

**Comment by tianhonglilei** [comment]
> 你好@tianhonglilei，您能否分享完整的错误日志并提出一个新的单独问题，并提供完整的重现步骤。谢谢！

Hi, thanks for your reply, the problem is solved, I replaced android_sdk version 30.0.1 to 30.0.3.
